mesh/mesh.cpp

```diff
@@ -6675,7 +6676,8 @@ int Mesh::CheckElementOrientation(bool fix_it)
                   wo++;
                   if (fix_it)
                   {
-                     // how?
+                     mfem::Swap(vi[1], vi[3]);
+                     fo++;
```

Comment:
For this to work we may need to change the `Finalize*` methods -- right now, it is assumed that only some mesh types (e.g. tet meshes) can be affected by this method, see https://github.com/mfem/mfem/blob/31a60b4476317ffc1b39421886fda798b2345de8/mesh/mesh.cpp#L3484-L3487

Similar change is implemented in PR #4607 for hexes and prisms, see the changes there in `Mesh::DoNodeReorder`, `Mesh::Finalize`, `Mesh::CheckElementOrientation`. In particular, for meshes with nodes, one probably needs to implement method `Mesh::GetPyramidOrientation` for the identity permutation and for the permutation that swaps vertices 1 and 3; this orientations then need to be supported in the H1 collection, see e.g. the changes in `H1_FECollection::DofOrderForOrientation` in PR #4607.